client/src/commons/Forms/PlacesTypesAndSubTypes/usePlacesTypesAndSubTypes.ts

```diff
@@ -7,15 +8,23 @@ import PlaceSubType from 'models/PlaceSubType';
 import PlacesSubTypesByTypes from 'models/PlacesSubTypesByTypes';
 import { setIsLoading } from 'redux/IsLoading/isLoadingActionCreators';
 
-import { usePlacesTypesAndSubTypesIncome } from './usePlacesTypesAndSubTypesInterfaces';
+
+import StoreStateType from 'redux/storeStateType';
+import {setPlaceTypes} from 'redux/PlaceTypes/placetypeActionCreators';
 
 const OTHER = 'אחר';
 
-const usePlacesTypesAndSubTypes = (parameters: usePlacesTypesAndSubTypesIncome) => {
+const usePlacesTypesAndSubTypes = () => {
+    const placesSubTypesByTypes = useSelector<StoreStateType,PlacesSubTypesByTypes>(state => state.placeSubTypesByTypes);
+
 
-    const { setPlacesSubTypesByTypes } = parameters;
+    const subtypesFetched = useMemo(() => Object.keys(placesSubTypesByTypes).length > 0, [placesSubTypesByTypes]);
 
     const getPlacesSubTypesByTypes = () => {
+        if (Object.keys(placesSubTypesByTypes).length > 0) {
```

Comment:
Maybe use Boolean(Object.keys(placesSubTypesByTypes)) and if it's false then do the rest of the stuff and not do return;

Reply:
im not sure i understand...

Reply:
also Boolean(Object.keys(placesSubTypesByTypes)) will always be true

Reply:
What i was trying to say is try checcking if there's length at all or length == 0, and not if there's length > 0 then return nothing
Maybe Boolean(Object.keys(placesSubTypesByTypes).length)